Documentation/saml-connector.md

```diff
@@ -0,0 +1,35 @@
+# Authentication through SAML
+
+## Overview
+
+This connector adds basic support for receiving authentication data through SAML. It uses a fork of `github.com/RobotsAndPencils/go-saml` hosted at `github.com/Calpicow/go-saml`.
+
+## Security considerations
+
+This connector uses `xmlsec` to perform signing/verification of the XML payload. There is currently an XXE vulnerability associated with libxml, which xmlsec uses. This is mitigated by rejecting XMLs that contain `<!DOCTYPE`.
+
+## Configuration
+
+The following is an example connector configuration for saml:
+
+```yaml
+connectors:
+- type: saml
+  id: saml
+  name: saml
+  config:
+    spIssuerUrl: $SP_ISSUER_URL
+    spSignRequest: false
+    # Typically urn:oasis:names:tc:SAML:2.0:nameid-format:transient or below
+    spNameIDPolicyFormat: urn:oasis:names:tc:SAML:2.0:nameid-format:persistent
+    # Set to empty string to omit RequestedAuthnContext block
+    spRequestedAuthnContext: urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport
+    idpSSOURL: $IDP_SSO_URL
+    idpPublicCertPath: /etc/dex/saml.crt
```

Comment:
"idp" and "sp" are probably redundant here. You can probably just use `issuerURL` and `ca`.